packages/apps/job-launcher/server/src/modules/cron-job/cron-job.service.ts

```diff
@@ -32,25 +32,21 @@ export class CronJobService {
   ) {}
 
   public async startCronJob(cronJobType: CronJobType): Promise<CronJobEntity> {
-    let cronJob = await this.cronJobRepository.findOne({
-      cronJobType,
-    });
+    let cronJob = await this.cronJobRepository.findOneByType(cronJobType);
```

Comment:
Let's use immutable variables with `const`.

```
const cronJob = await this.cronJobRepository.findOneByType(cronJobType);

if (!cronJob) {
  return this.cronJobRepository.createUnique(cronJobType);
}

cronJob.startedAt = new Date();
cronJob.completedAt = null;

return this.cronJobRepository.updateOne(cronJob);
```

Reply:
@portuu3

Reply:
We can't do it that way. With the example you have provided, you are not assigning any values to the startedAt and completedAt properties as the original logic did. Anyways, it could be done by adding to the entity properties the default values @eugenvoronov

Reply:
But as I can see its totally the same logic as you did, no?
To use a default value its a good point @flopez7